pkg/cli/testdata/dump_test

```diff
@@ -0,0 +1,102 @@
+% name simple_dump
+% command dump d t
+CREATE DATABASE d;
+CREATE TABLE d.t (
+  a INT
+);
+INSERT INTO d.t VALUES
+  (1);
+----
+dump d t
+CREATE TABLE t (
+	a INT NULL,
+	FAMILY "primary" (a, rowid)
+);
+
+INSERT INTO t (a) VALUES
+	(1);
+
+% name inverted_index
+% command dump d t
+CREATE DATABASE d;
+CREATE TABLE d.t (
+	a JSON,
+	b JSON,
+	INVERTED INDEX idx (a)
+);
+
+CREATE INVERTED INDEX idx2 ON d.t (b);
+
+INSERT INTO d.t VALUES ('{"a": "b"}', '{"c": "d"}');
+----
+dump d t
+CREATE TABLE t (
+	a JSON NULL,
+	b JSON NULL,
+	INVERTED INDEX idx (a),
+	INVERTED INDEX idx2 (b),
+	FAMILY "primary" (a, b, rowid)
+);
+
+INSERT INTO t (a, b) VALUES
+	('{"a":"b"}', '{"c":"d"}');
+
+% name reference_cycle
+% command dump d t
+-- This tests dumping in the presence of cycles.  This used to crash before
```

Comment:
I would suggest using # instead of -- for comments, like we do in the logic tests.